model-optimizer/unit_tests/mo/graph/graph_test.py

```diff
@@ -1083,6 +1156,140 @@ def test_node_in_out_ports_order(self):
+                             ('op_split', 'op_split_4_data', {'out': 'control_flow_4', 'control_flow_edge': True}),
+                             ('op_split', 'op_split_5_data', {'out': 'control_flow_5', 'control_flow_edge': True}),
+                             ('op_split', 'op_split_6_data', {'out': 'control_flow_6', 'control_flow_edge': True}),
+                             ('op_split', 'op_split_7_data', {'out': 'control_flow_7', 'control_flow_edge': True}),
+                             ('op_split', 'op_split_8_data', {'out': 'control_flow_8', 'control_flow_edge': True}),
+                             ('op_split', 'op_split_9_data', {'out': 'control_flow_9', 'control_flow_edge': True}),
+                             ('op_split', 'op_split_10_data', {'out': 'control_flow_10', 'control_flow_edge': True}),
+                             ('op_split_0_data', 'out_0', {'in': 'control_flow_0', 'control_flow_edge': True}),
+                             ('op_split_1_data', 'out_1', {'in': 'control_flow_0', 'control_flow_edge': True}),
+                             ('op_split_2_data', 'out_2', {'in': 'control_flow_0', 'control_flow_edge': True}),
+                             ('op_split_3_data', 'out_3', {'in': 'control_flow_0', 'control_flow_edge': True}),
+                             ('op_split_4_data', 'out_4', {'in': 'control_flow_0', 'control_flow_edge': True}),
+                             ('op_split_5_data', 'out_5', {'in': 'control_flow_0', 'control_flow_edge': True}),
+                             ('op_split_6_data', 'out_6', {'in': 'control_flow_0', 'control_flow_edge': True}),
+                             ('op_split_7_data', 'out_7', {'in': 'control_flow_0', 'control_flow_edge': True}),
+                             ('op_split_8_data', 'out_8', {'in': 'control_flow_0', 'control_flow_edge': True}),
+                             ('op_split_9_data', 'out_9', {'in': 'control_flow_0', 'control_flow_edge': True}),
+                             ('op_split_10_data', 'out_10', {'in': 'control_flow_0', 'control_flow_edge': True}),
+                             ])
+
+        node_concat = Node(graph, 'op_concat')
+        node_split = Node(graph, 'op_split')
+
+        self.assertEqual(len(node_concat.in_ports()), len(node_concat.in_nodes()))
+
+        l1 = [node_concat.in_port(idx, control_flow=True).get_source().node.name
+              for idx in node_concat.in_ports(control_flow=True)]
+        l2 = [node_concat.in_node(idx, control_flow=True).in_node(0, control_flow=True).name
+              for idx in node_concat.in_nodes(control_flow=True)]
+
+        self.assertEqual(l1, l2)
+
+        l1 = [node_split.out_port(idx, control_flow=True).get_destination().node.name
+              for idx in node_split.out_ports(control_flow=True)]
+        l2 = [node_split.out_node(idx, control_flow=True).out_node(0, control_flow=True).name for idx in
+              node_split.out_nodes(control_flow=True)]
+
+        self.assertEqual(l1, l2)
+
 
```

Comment:
please add one more test case (or substitute the first two cases) with both data and control flow ports.

Reply:
New test with both data and control flow ports was added.